src/commands/serve/serve.mjs

```diff
@@ -0,0 +1,182 @@
+// @ts-check
+import process from 'process'
+
+import { Option } from 'commander'
+
+import { promptEditorHelper } from '../../lib/edge-functions/editor-helper.mjs'
+import { startFunctionsServer } from '../../lib/functions/server.mjs'
+import { printBanner } from '../../utils/banner.mjs'
+import {
+  chalk,
+  exit,
+  log,
+  NETLIFYDEVERR,
+  NETLIFYDEVLOG,
+  NETLIFYDEVWARN,
+  normalizeConfig,
+} from '../../utils/command-helpers.mjs'
+import detectServerSettings, { getConfigWithPlugins } from '../../utils/detect-server-settings.mjs'
+import { getSiteInformation, injectEnvVariables } from '../../utils/dev.mjs'
+import { getEnvelopeEnv, normalizeContext } from '../../utils/env/index.mjs'
+import { ensureNetlifyIgnore } from '../../utils/gitignore.mjs'
+import openBrowser from '../../utils/open-browser.mjs'
+import { generateInspectSettings, startProxyServer } from '../../utils/proxy-server.mjs'
+import { runBuildTimeline } from '../../utils/run-build.mjs'
+import { getGeoCountryArgParser } from '../../utils/validation.mjs'
+
+/**
+ * The serve command
+ * @param {import('commander').OptionValues} options
+ * @param {import('../base-command.mjs').default} command
+ */
+const serve = async (options, command) => {
+  const { api, cachedConfig, config, repositoryRoot, site, siteInfo, state } = command.netlify
+  config.dev = { ...config.dev }
+  config.build = { ...config.build }
+  /** @type {import('../dev/types').DevConfig} */
+  const devConfig = {
+    framework: '#auto',
+    ...(config.functionsDirectory && { functions: config.functionsDirectory }),
+    ...(config.build.publish && { publish: config.build.publish }),
+    ...config.dev,
+    ...options,
+  }
+
+  let { env } = cachedConfig
+
+  // Override the `framework` value so that we start a static server and not
+  // the framework's development server.
+  devConfig.framework = '#static'
```

Comment:
```suggestion
  const devConfig = {
    ...(config.functionsDirectory && { functions: config.functionsDirectory }),
    ...(config.build.publish && { publish: config.build.publish }),
    ...config.dev,
    ...options,
    // Override the `framework` value so that we start a static server and not
    // the framework's development server.  
    framework: '#static',
  }

  let { env } = cachedConfig
```

Can we do this instead?

Reply:
Absolutely!